api/envoy/config/core/v3/socket_option.proto

```diff
@@ -51,6 +51,26 @@ message SocketOption {
     STATE_LISTENING = 2;
   }
 
+  // The `socket type <https://linux.die.net/man/2/socket>`_ to apply the socket option to.
+  // If :ref:`stream <envoy_v3_api_field_config.core.v3.SocketOption.SocketType.stream>` is set,
+  // it takes precedence over :ref:`datagram <envoy_v3_api_field_config.core.v3.SocketOption.SocketType.datagram>`.
+  // If none of the fields is set, the socket option will be applied to all socket types.
+  message SocketType {
```

Comment:
Consider clarifying the comment explaining that up to one should be provided, and if multiple are provided then the precedence order will determine the chosen one.

Reply:
Done.